### DISCUSSION TITLE
Error when trying to request to api.tidal.com

### DISCUSSION THREAD

**Discussion by SergioFdezRc** [tidal-music/discussions#78] [Q&A]
Hi @stefanmoro,

I have done the authentication with Tidal using the Web API guide, using the endpoint that returns you the authorization code (`https://login.tidal.com/authorize`) using these scopes: 

- user.read
- playlists.read
- playlists.write
- collection.read
- collection.write

Then I requested the exchange authorization code for token (`https://auth.tidal.com/v1/oauth2/token`) and logged in without problems.

However, when I try to make a request to the following url:

`https://api.tidal.com/v1/users/${user_id}?output=json&countryCode=US`, with the **access_token** in the `Authorization` entry in the `header`.

The request returns the following error:

```
{
    "status": 403,
    "subStatus": 11004,
    "userMessage": "Token is missing required scope. Required scopes: r_usr Actual scopes: user.read collection.read playlists.write collection.write playlists.read"
}
```

If I try to set the `r_usr` scope when requesting the authentication code, the authentication process fails. Is the `api.tidal.com` endpoint restricted for use?

Thanks in advance,

Sergio.

**Comment by stefanmoro** [comment]
Hey @SergioFdezRc ,

Yes, the only endpoints that are available for use are the ones documented at https://developer.tidal.com/reference/web-api (excluding OAuth endpoints). Other functionality, like playback is exclusively available via our SDKs.

You can see the scopes you are allowed to acquire in the developer portal dashboard for your app.

We are actively opening up new endpoints now, and will continue doing so throughout the next few months.

**Reply by SergioFdezRc** [reply]
Thanks for your quick response, @stefanmoro. I want to get the playlists for my user and I can't find that endpoint anywhere. 

Is this functionality available or will it be in the near future? Our company is developing a very interesting product that needs to use this functionality, if not, it would be a shame not to be able to use it.

**Reply by stefanmoro** [reply]
We are working on playlists right now, and we aim to have read access released within the next month, and write access shortly thereafter.

In addition to playlists, we are also working on:
User info - [released today](https://developer.tidal.com/reference/web-api?spec=user-v2&ref=get-user-v2)
My collection
Recommendations

We will roll out new endpoints in beta as soon as they are ready "enough".

**Reply by SergioFdezRc** [reply]
Thank you for your response. In that case, we will be waiting for this implementation.

Thank you very much, @stefanmoro

**Reply by stefanmoro** [reply]
Great! We'll announce in this discussions forum as we make progress.

**Comment by cadamini** [comment]
For everyone still looking for a solution as of the end of 2025, I ended up in the same situation with the v2 beta API and the required `r_usr` scope.

I found the Python-based [spotify_to_tidal ](https://github.com/spotify2tidal/spotify_to_tidal?tab=readme-ov-file) (using SDKs/v1 APIs). The tool worked like a charm.
